### DISCUSSION TITLE
SPF: HELO does not publish an SPF Record

### DISCUSSION THREAD

**Discussion by impirecom** [postalserver/postal#2555] [Help with using Postal]
Hello community,

This seems to be an issue with older email servers/spam checkers still looking for that perfect header. 

My server have a pool of 35 IPs. Reverse PTR records all pointed to the server name postal.example.com. All passed rDNS tests.

I used mail-tester.com to test a send email. If the sent email come from the same main IP that the server is bind to, it passed with a perfect score. But if the sent IP is different than the main server IP, then the score dropped and I get the below errors

"-0.1 | DKIM_SIGNED | Message has a DKIM or DK signature, not necessarily validThis rule is automatically applied if your email contains a DKIM signature but other positive rules will also be added if your DKIM signature is valid. See immediately below.
-- | -- | --
0.1 | DKIM_VALID | Message has at least one valid DKIM or DK signatureGreat! Your signature is valid
0.1 | DKIM_VALID_AU | Message has a valid DKIM or DK signature from author's domainGreat! Your signature is valid and it's coming from your domain name
0.1 | DKIM_VALID_EF | Message has a valid DKIM or DK signature from envelope-from domain
-1.274 | RDNS_NONE | Delivered to internal network by a host with no rDNSThis may indicate you do not have a rDNS configured for your hostname or the rDNS does not match your sending IP
-0.001 | SPF_HELO_NONE | SPF: HELO does not publish an SPF Record
0.001 | SPF_PASS | SPF: sender matches SPF recordGreat! Your SPF is valid"

Any suggestions on how to solve this issue? I read that by adding all IPs into the SPF would be too taxing on the DNS lookups. But adding A records for each of these IP just doesn't work.

Thanks in advance for your help.

Kind regards,

Andy

<img width="1484" alt="Screenshot 2023-07-22 at 1 38 13 PM" src="https://github.com/postalserver/postal/assets/79177007/c752dff8-eaa5-40b7-bde4-5f4078e22ac0">

**Comment by willpower232** [comment]
You can see in the list that the SPF (and DKIM) results cancel each other out and the main problem is that it could not validate your reverse DNS.

If you have set up the PTR, it might be worth verifying that they are actually set up correctly using other services.

**Reply by impirecom** [reply]
Thank you very much @willpower232 

Do you have a sample DNS records for a proper SPF and DKIM setup to avoid this problem?

**Reply by willpower232** [reply]
As I said, I don't think the problem is SPF or DKIM, the major issue seems to be the reverse DNS.

**Reply by impirecom** [reply]
Thank you @willpower232. Here's the setup:

Server name: postal.example.com, bounded to 80.80.80.1
Server IP pool: 80.80.80.1-100
PTR record (reverse DNS) for IP range 80.80.80.1-200 all pointed to postal.example.com

If mails get send out on 80.80.80.1, will pass mail-tester.com with perfect 10.
If mails get send out on 80.80.80.2-100 will not get the perfect score. Instead, I get the mentioned error as you seen above?

What is your suggestion to fix this? I just don't know what else to do on the PTR records except for making sure they all pointed to the mail server which all IPs are configured for that server.

**Reply by willpower232** [reply]
mxtoolbox.com has a reverse IP lookup so if you enter the other IP addresses in that form, does that confirm whether the reverse DNS has been applied properly?

**Reply by impirecom** [reply]
Yes. The IP addresses matched the main server name, postal.example.com and passed reverse DNS lookup. But problem is unless an email is sent out via that same default IP address which Postal server bound to, 80.80.80.1, it will not pass with perfect score 10

**Reply by willpower232** [reply]
If you're happy with the reverse DNS records then its entirely possible this is a false negative. Do you have issues sending emails?

**Reply by impirecom** [reply]
No issue sending. Just not getting a 10 on several email testers.
